src/main/java/io/cdap/plugin/cdc/source/sqlserver/ResultSetToDMLRecord.java

```diff
@@ -40,57 +40,120 @@
  * to {@link StructuredRecord} for dml records
  */
 public class ResultSetToDMLRecord implements Function<ResultSet, StructuredRecord> {
-  private static final int CHANGE_TABLE_COLUMNS_SIZE = 3;
+  private static final Logger LOG = LoggerFactory.getLogger(ResultSetToDMLRecord.class);
+
+  private static final int CHANGE_TABLE_COLUMNS_SIZE_WITHOUT_SQN = 3;
+  private static final int CHANGE_TABLE_COLUMNS_SIZE_WITH_SQN = 2;
+  private static int size = CHANGE_TABLE_COLUMNS_SIZE_WITHOUT_SQN;
   private final TableInformation tableInformation;
+  private final boolean requireSeqNumber;
 
-  ResultSetToDMLRecord(TableInformation tableInformation) {
+  ResultSetToDMLRecord(TableInformation tableInformation, boolean requireSeqNumber) {
+    this.requireSeqNumber = requireSeqNumber;
     this.tableInformation = tableInformation;
+    if (requireSeqNumber) {
+      size = CHANGE_TABLE_COLUMNS_SIZE_WITH_SQN;
+    }
   }
 
   @Override
-  public StructuredRecord call(ResultSet row) throws SQLException {
-    Schema changeSchema = getChangeSchema(row);
-    String operation = row.getString("SYS_CHANGE_OPERATION");
-    OperationType operationType = OperationType.fromShortName(operation);
+  public StructuredRecord call(ResultSet row) throws Exception {
+    LOG.info(row.toString());
+    Schema changeSchema = getChangeSchema(row, size);
+    Map<String, Object> map = getChangeData(row, changeSchema, size);
     return StructuredRecord.builder(Schemas.DML_SCHEMA)
-      .set(Schemas.TABLE_FIELD, Joiner.on(".").join(tableInformation.getSchemaName(), tableInformation.getName()))
+      .set(Schemas.TABLE_FIELD, Joiner.on(".").join(tableInformation.getSchemaName(),
+        tableInformation.getName()))
       .set(Schemas.PRIMARY_KEYS_FIELD, Lists.newArrayList(tableInformation.getPrimaryKeys()))
-      .set(Schemas.OP_TYPE_FIELD, operationType.name())
+      .set(Schemas.OP_TYPE_FIELD, getChangeOperation(row).name())
       .set(Schemas.UPDATE_SCHEMA_FIELD, changeSchema.toString())
-      .set(Schemas.UPDATE_VALUES_FIELD, getChangeData(row, changeSchema))
+      .set(Schemas.UPDATE_VALUES_FIELD, map)
+      .set(Schemas.CHANGE_TRACKING_VERSION, row.getString("CHANGE_TRACKING_VERSION"))
       .build();
   }
 
-  private static Map<String, Object> getChangeData(ResultSet resultSet, Schema changeSchema) throws SQLException {
+  private static OperationType getChangeOperation(ResultSet row) throws Exception {
```

Comment:
why the change to not use OperationType.fromShortName()? The logic should be encapsulated in the enum, and not leaked out to this class.

Reply:
agreed